jdaviz/configs/imviz/plugins/coords_info/coords_info.py

```diff
@@ -473,11 +473,24 @@ def _image_viewer_update(self, viewer, x, y):
                 value = self._get_cube_value(
                     image, arr, x, y, viewer
                 )
-                if self.image_unit is not None and self.image_unit.is_equivalent(unit):
-                    value = _convert_surface_brightness_units(
-                        value, unit, self.image_unit
-                    )
-                    unit = self.image_unit
+                if self.image_unit is not None:
+                    if 'PIXAR_SR' in self.app.data_collection[0].meta:
+                        # Need current slice value and associated unit to use to compute
+                        # spectral density equivalencies that enable Flux to Flux conversions.
+                        # This is needed for units that are not directly convertible/translatable.
+                        slice = viewer.slice_value * u.Unit(self.app._get_display_unit('spectral'))
+
+                        value = flux_conversion(values=value, original_units=unit,
+                                                target_units=self.image_unit, spec=None,
+                                                eqv=_eqv_pixar_sr(self.app.data_collection[0].meta['PIXAR_SR']),  # noqa
+                                                slice=slice)
+                        unit = self.image_unit
+
+                    elif self.image_unit.is_equivalent(unit):
+                        value = _convert_surface_brightness_units(
+                            value, unit, self.image_unit
+                        )
```

Comment:
Given this function is only 2 lines (and really those 2 lines are generic and can be shortened to one-liner, not specific to surface brightness):

```python
def _convert_surface_brightness_units(data, original_units, target_units):
    quantity = data * u.Unit(original_units)
    return quantity.to_value(u.Unit(target_units))
```

```suggestion
                        value = (value * u.Unit(unit)).to_value(u.Unit(self.image_unit))
```

🤷‍♀️

Reply:
That works for me! All I did in the PR was change this from the primarily conversion of surface brightness in image-viewer's mouseover to the fallback and update the parameter language to match it with the `flux_conversion` syntax.